Documentation/bpf.rst

```diff
@@ -3354,7 +3354,355 @@ the toolchain section under the respective tools.
 tc (traffic control)
 --------------------
 
-TODO
+Aside from other program types such as XDP, BPF can also be used out of the
+kernel's tc (traffic control) layer in the networking data path. On a high-level
+there are three major differences when comparing BPF XDP programs to tc BPF
+ones:
+
+* The BPF input context is a ``sk_buff`` not a ``xdp_buff``. The former is
+  the network packet representation of the kernel's networking stack and
+  therefore used in many places. As such it has all the relevant ``sk_buff``
+  metadata available the BPF program can read and write into, whereas in
+  ``xdp_buff`` case it's limited due to the early stage processing where the
+  cost of allocating a ``sk_buff`` is avoided altogether.
+
+  Out of the tc BPF hooks, the BPF program can for instance read or write
```

Comment:
Perhaps,

`Out of the tc BPF hooks, the BPF program can...`
=>
`BPF programs attached to the tc BPF hook can...`